### PR TITLE
Publish ES module

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
The main reason for this PR and new major version is to dual publish the library as an ES module alongside the existing bundle, so it can more efficiently consumed and bundled in upstream projects which use ESM. To achieve this we've moved from Webpack to Rollup for the compilation and bundling. 

We also took the opportunity to clean up the project, bump the TypeScript version, remove Babel dependency (and thus rely on the TS internal libraries) and fix some tests. Due to the new bundle strategy (Rollup) and compilation (remove Babel) this warrants the bump to a new major version. However, no core functionality has been removed or added in this change and has been canaried for a long period of time in Fastly systems to gain confidence in the change.

### What?
- Move form Webpack -> Rollup for module bundling
- Remove Babel support
- Export ESM module and reference in package.json
- Upgrade TypeScript and fix breaking tests from upgrade
- Replace TravisCI with GitHub actions

### CODE REVIEW

**Comment by doramatadora** [comment]
Having tested this locally, I'm delighted with the new bundling experience 🚀

**Review by doramatadora — APPROVED** [review]
